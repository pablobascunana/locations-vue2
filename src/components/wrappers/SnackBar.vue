<template>
  <v-snackbar
    :color="color"
    :top="true"
    v-model="showSnackBar"
  >
    {{ message }}
  </v-snackbar>
</template>

<script>
export default {
  name: "snackBar",
  props: {
    color: { type: String, required: true },
    message: { type: String, required: true }
  },
  data() {
    return {
      showSnackBar: true
    }
  },
  mounted() {
    this.closeSnackBar();
  },
  methods: {
    closeSnackBar() {
      setTimeout(() => {
        this.showSnackBar = false;
        this.$emit('close');
      }, 8000);
    }
  }
}
</script>
