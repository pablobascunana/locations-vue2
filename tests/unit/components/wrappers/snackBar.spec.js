import { createLocalVue, shallowMount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import SnackBar from '@/components/wrappers/SnackBar.vue'

const localVue = createLocalVue();
let vuetify = new Vuetify();


describe('App', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(SnackBar, {
            localVue,
            vuetify,
        });
    });

    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Call to close snackBar', () => {
        jest.useFakeTimers();
        wrapper.vm.closeSnackBar();
        jest.advanceTimersByTime(8000);
        expect(wrapper.vm.showSnackBar).toBe(false);
    });
});