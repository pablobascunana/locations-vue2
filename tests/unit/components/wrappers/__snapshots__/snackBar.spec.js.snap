// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<v-snackbar top="true">

</v-snackbar>
`;
